account_move_default_journal/models/account_move.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def _search_default_journal(self, journal_types):
+        company_id = self._context.get("default_company_id", self.env.company.id)
+        domain = [
+            ("company_id", "=", company_id),
+            ("type", "in", journal_types),
+        ]
+        if self._context.get("default_currency_id"):
+            domain.append(("currency_id", "=", self._context["default_currency_id"]))
+
+        # Use the appropriate config parameter for the journal type
+        journal_id = None
+        if "sale" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.sale_journal_id"
+            )
+        elif "purchase" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.purchase_journal_id"
+            )
+        elif "general" in journal_types:
+            journal_id = self.env["ir.config_parameter"].get_param(
+                "account_move_default_journal.general_journal_id"
+            )
+
+        # Try to get the journal, or default back to normal behavior
+        if journal_id:
+            journal_id = int(journal_id)
+            journal = self.env["account.journal"].search(
```

Comment:
Note that this might result in an empty record set. If you apply the other changes I suggested you do not need a search at all, therefore you could dispense with creating a domain altogether as well.